app/models/service_template_provision_request.rb

```diff
@@ -57,6 +57,25 @@ def requested_task_idx
 
   def customize_request_task_attributes(req_task_attrs, idx)
     req_task_attrs['options'][:pass] = idx
+
+    configuration_script_id = resource_action&.configuration_script_id
+    return if configuration_script_id.nil?
+
+    # If the service_template that we are provisioning is a "generic" provision
+    # type then we want to execute the request task with embedded_workflows if
+    # the resource_action has a configuration_script_id
+    if source.prov_type == "generic"
+      req_task_attrs['options'][:configuration_script_payload_id] = configuration_script_id
+    else
+      req_task_attrs['options'][:parent_configuration_script_payload_id] = configuration_script_id
+    end
+  end
+
+  def resource_action
+    resource_action_id = options.dig(:workflow_settings, :resource_action_id)
+    return if resource_action_id.nil?
+
+    ResourceAction.find(resource_action_id)
```

Comment:
I like the shorter format, but both work:

```suggestion
    ResourceAction.find(resource_action_id) if resource_action_id.nil?
```